iconfetcher.go

```diff
@@ -0,0 +1,108 @@
+package jujusvg
+
+import (
+	"io/ioutil"
+	"net/http"
+	"sync"
+
+	"gopkg.in/errgo.v1"
+	"gopkg.in/juju/charm.v5-unstable"
+)
+
+// An IconFetcher provides functionality for retrieving icons for the charms
+// within a given bundle.  The FetchIcons function accepts a bundle, and
+// returns a dict mapping charm paths to icon data.
+type IconFetcher interface {
+	FetchIcons(*charm.BundleData) (map[string]string, error)
+}
+
+// HttpFetcher is a default implementation of IconFetcher which retrieves charm
+// icons from the web using the URL generated by IconURL on that charm.  The
+// HTTP Client used may be overridden by an instance of http.Client.  The icons
+// may optionally be fetched concurrently.
+type HttpFetcher struct {
+	FetchConcurrently bool
+	IconURL           func(*charm.Reference) string
```

Comment:
// IconURL returns the URL from which to
    // fetch the given entity id.

?